app/src/main/java/io/github/sustainow/presentation/ui/components/NumericalSelectQuestionCard.kt

```diff
@@ -0,0 +1,141 @@
+import kotlin.time.Duration
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.runtime.saveable.Saver
+
+@Composable
+fun NumericalSelectQuestionCard(
+    question: Question.Numerical,
+    onAlternativeSelected: (QuestionAlternative) -> Unit,
+) {
+    // Defina um Saver personalizado para TextFieldValue
+    val textFieldValueSaver = Saver<TextFieldValue, String>(
+        save = { it.text }, // Salva apenas o texto do TextFieldValue
+        restore = { TextFieldValue(it) } // Restaura o TextFieldValue a partir do texto salvo
+    )
+
+    var textFieldValue by rememberSaveable(stateSaver = textFieldValueSaver) {
+        mutableStateOf(TextFieldValue())
+    }
+
+    Card(
+        modifier = Modifier
+            .width(360.dp)
+            .padding(vertical = 10.dp)
+            .shadow(
+                elevation = 8.dp,
+                shape = RoundedCornerShape(0.dp),
+            )
+            .padding(horizontal = 10.dp),
+        colors = CardDefaults.cardColors(
+            containerColor = onSurfaceDarkHighContrast, // Cor do fundo do Card
+        ),
+    ) {
+        Column(
+            modifier = Modifier
+                .padding(10.dp)
+                .fillMaxWidth()
+                .wrapContentHeight(),
+        ) {
+            // Cabeçalho com nome e texto da questão
+            Text(
+                text = question.name,
+                style = MaterialTheme.typography.titleMedium.copy(color = scrimLight),
+                modifier = Modifier.padding(bottom = 4.dp),
+            )
+            Text(
+                text = question.text,
+                style = MaterialTheme.typography.bodyMedium.copy(color = scrimLight),
+                modifier = Modifier.padding(bottom = 8.dp),
+            )
+
+            HorizontalDivider(
+                modifier = Modifier.padding(bottom = 10.dp),
+                thickness = 1.dp,
+                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f),
+            )
+
+            // Campo de texto numérico
+            BasicTextField(
+                value = textFieldValue,
+                onValueChange = { newValue ->
+                    textFieldValue = newValue
+                    val alternative = question.alternatives.firstOrNull()
+                    if (alternative != null) {
+                        onAlternativeSelected(
+                            alternative.copy(value = newValue.text.toFloatOrNull() ?: 0f)
+                        )
+                    }
+                },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(vertical = 10.dp)
+                    .background(color = onBackgroundDarkHighContrast),
+                textStyle = MaterialTheme.typography.bodyLarge.copy(color = scrimLight),
+                singleLine = true,
+            )
```

Comment:
Adicionar keyboardOptions= KeyboardOptions(keyboardType = KeyboardType.Number), para permitir apenas números.
utilizar toFloat em vez de toFloat or null e quando uma NumberFormatException for capturada refletir no atributo isError do textfield e mostrar uma mensagem de erro no atributo supporting text (lembrar de usar recursos de string)